toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -247,8 +250,37 @@ func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string,
 	)
 	defer func() {
 		if err != nil {
-			span.RecordError(fmt.Errorf("image customization failed"))
-			span.SetStatus(codes.Error, "image customization failed")
+			category := GetErrorCategory(err)
+			code := GetErrorCode(err)
+			
+			// Create JSON structure for error details
+			errorDetails := map[string]string{
```

Comment:
rather than doing all of this logic, we should have a function in errors.go that gives us the json representation of the error with just the category and code (since the error message itself might have PII), which we can then use for RecordError().